File: src/main/java/cat/xlagunas/andrtc/push/PushNotificationRepository.java
package cat.xlagunas.andrtc.push;

import java.util.concurrent.CompletableFuture;

import okhttp3.ResponseBody;
import retrofit2.http.Body;
import retrofit2.http.POST;

public interface PushNotificationRepository {

    @POST("/fcm/send")
    CompletableFuture<ResponseBody> sendPush(@Body PushMessage message);
}

File: src/main/java/cat/xlagunas/andrtc/push/PushNotificationRepository.kt
package cat.xlagunas.andrtc.push

import okhttp3.ResponseBody
import retrofit2.http.Body
import retrofit2.http.POST
import java.util.concurrent.CompletableFuture

interface PushNotificationRepository {
    @POST("/fcm/send")
    fun sendPush(@Body message: PushMessage): CompletableFuture<ResponseBody>
}
